crescent/commands/options.py

```diff
@@ -0,0 +1,225 @@
+    max_value: Optional[Union[int, float]]
+
+    def _gen_option(self, name: str) -> CommandOption:
+        return CommandOption(
+            type=self.type,
+            name=name,
+            description=self.description,
+            is_required=self.default is UNDEFINED,
+            choices=self.choices,
+            channel_types=self.channel_types,
+            min_value=self.min_value,
+            max_value=self.max_value,
+        )
+
+
+D = TypeVar("D")
+IF = TypeVar("IF", int, float)
+ROMB = TypeVar("ROMB", User, Role, Mentionable, bool)
+
+
+@overload
+def option(
+    type: Union[Type[PartialChannel], Sequence[PartialChannel]],
+    description: str = ...,
+) -> InteractionChannel:
+    ...
+
+
+@overload
+def option(
+    type: Union[Type[PartialChannel], Sequence[PartialChannel]],
+    description: str = ...,
+    *,
+    default: D,
+) -> Union[InteractionChannel, D]:
+    ...
+
+
+@overload
+def option(
+    type: Type[ROMB],
+    description: str = ...,
+) -> ROMB:
+    ...
+
+
+@overload
+def option(
+    type: Type[ROMB],
+    description: str = ...,
+    *,
+    default: D,
+) -> Union[ROMB, D]:
+    ...
+
+
+@overload
+def option(
+    type: Type[IF],
+    description: str = ...,
+    *,
+    choices: Optional[Sequence[Tuple[str, IF]]] = ...,
+    min_value: Optional[IF] = ...,
+    max_value: Optional[IF] = ...,
+) -> IF:
+    ...
+
+
+@overload
+def option(
+    type: Type[IF],
+    description: str = ...,
+    *,
+    default: D,
+    choices: Optional[Sequence[Tuple[str, IF]]] = ...,
+    min_value: Optional[IF] = ...,
+    max_value: Optional[IF] = ...,
+) -> Union[IF, D]:
+    ...
+
+
+@overload
+def option(
+    type: Type[str],
+    description: str = ...,
+    *,
+    choices: Optional[Sequence[Tuple[str, str]]] = ...,
+    min_value: Optional[str] = ...,
+    max_value: Optional[str] = ...,
+) -> str:
+    ...
+
+
+@overload
+def option(
+    type: Type[str],
+    description: str = ...,
+    *,
+    default: D,
+    choices: Optional[Sequence[Tuple[str, str]]] = ...,
+    min_value: Optional[str] = ...,
+    max_value: Optional[str] = ...,
+) -> Union[str, D]:
+    ...
+
+
+def option(  # type: ignore
+    ctype: Union[Type[_VALID_TYPES], Sequence[Type[_VALID_CHANNEL_TYPES]]],
+    description: str = "\u200B",
+    *,
+    default: UndefinedOr[Any] = UNDEFINED,
+    choices: Sequence[Tuple[str, Union[str, int, float]]] | None = None,
+    min_value: Optional[Union[int, float]] = None,
+    max_value: Optional[Union[int, float]] = None,
+) -> Any:
+    if isinstance(ctype, type) and issubclass(ctype, PartialChannel):
+        if ctype is PartialChannel:
+            channel_types = None
+        else:
+            channel_types = _get_channel_t(ctype)
+        ctype = PartialChannel
+    elif isinstance(ctype, Sequence):
+        channel_types = []
+        for t in ctype:
+            channel_types.extend(_get_channel_t(t))
+        ctype = PartialChannel
+    else:
+        channel_types = None
```

Comment:
I think this logic should be moved to a separate function it can be used to resolve types for Annotated command options

Also grab the `OPTIONS_TYPE_MAP` part